Changes for rsc app installation support in v1.0 (#21605)
microsoftgraph/microsoft-graph-docs: api-reference/v1.0/api/team-teamsappinstallation-upgrade.md

---

> title: "teamsAppInstallation in team: upgrade"

Reviewer: ```suggestion
title: "teamsAppInstallation in a team: upgrade"
```

description: "Upgrade an app installation in a team"
author: "akjo"
ms.localizationpriority: medium
ms.prod: "microsoft-teams"
doc_type: apiPageType
---

> # teamsAppInstallation in team: upgrade

Reviewer: ```suggestion
# teamsAppInstallation in a team: upgrade
```

Namespace: microsoft.graph

Upgrade an [app installation](../resources/teamsappinstallation.md) in a [team](../resources/team.md) to the latest version of the app.

> **Note:**

> > - Currently, upgrading an app that requires resource-specific consent permissions is not supported in application context.

Reviewer: ```suggestion
> - Currently, upgrading an app that requires resource-specific consent permissions is not supported in an application context.
```

> > - Currently, upgrading an app that requires resource-specific consent permissions is not supported in an application context.

Reviewer: ```suggestion
> Currently, upgrading an app that requires resource-specific consent permissions is not supported in an application context.
```

## Permissions

One of the following permissions is required to call this API. To learn more, including how to choose permissions, see [Permissions](/graph/permissions-reference).

|Permission type      | Permissions (from least to most privileged)              |
|:--------------------|:---------------------------------------------------------|

> |Delegated (work or school account) | TeamsAppInstallation.ReadWriteSelfForTeam<sup>1</sup>, TeamsAppInstallation.ReadWriteForTeam<sup>1</sup>, TeamsAppInstallation.ReadWriteAndConsentForTeam, TeamsAppInstallation.ReadWriteAndConsentSelfForTeam, Group.ReadWrite.All<sup>2</sup>, Directory.ReadWrite.All<sup>2</sup> |

Reviewer: Are these new permissions? If so, please add changelog entries.

## Request body

In the request body, supply a JSON representation of the parameters.

The following table shows additional parameters that can be used with the upgrade action.

|Parameter|Type|Description|
|:---|:---|:---|
|consentedPermissionSet|[teamsAppPermissionSet](../resources/teamsapppermissionset.md)|Set of resource specific permissions that are being consented.|

> **Note**:

> > The permissions consented to during the upgrade must be the same as the resource-specific permissions present in the [teamsAppDefinition](../resources/teamsAppDefinition.md) of the app. To get the application and delegated resource-specific permissions, see [Example 7](../api/appcatalogs-list-teamsapps.md#example-7-list-applications-with-a-given-id-and-return-only-the-resource-specific-permissions-required-by-the-app). If only delegated resource-specific permissions are present in **teamsAppDefinition**, permissions can be omitted in the body of this request.

Reviewer: ```suggestion
> The permissions consented to during the installation must match the resource-specific permissions defined in the [teamsAppDefinition](../resources/teamsAppDefinition.md) of the app. To get the application and delegated resource-specific permissions, see [Example 7](../api/appcatalogs-list-teamsapps.md#example-7-list-applications-with-a-given-id-and-return-only-the-resource-specific-permissions-required-by-the-app). If only delegated resource-specific permissions are present in **teamsAppDefinition**, permissions can be omitted in the body of this request.
```
